recipe/build_pytorch.sh

```diff
@@ -108,6 +109,8 @@ if [[ ${cuda_compiler_version} != "None" ]]; then
     fi
     export TORCH_NVCC_FLAGS="-Xfatbin -compress-all"
     export NCCL_ROOT_DIR=$PREFIX
+    export NCCL_INCLUDE_DIR=$PREFIX/include
+    export USE_SYSTEM_NCCL=1
```

Comment:
There were some issues previously when building gloo together with system nccl (https://github.com/pytorch/pytorch/issues/36570, https://github.com/pytorch/pytorch/issues/14537).

Judging from the log it appears that gloo correctly found the conda-forge nccl
```
2021-08-01T18:17:15.7887128Z   -- Gloo build as SHARED library
2021-08-01T18:17:15.7954887Z   -- Found CUDA: /usr/local/cuda (found suitable version "10.2", minimum required is "7.0")
2021-08-01T18:17:15.7969934Z   -- CUDA detected: 10.2
2021-08-01T18:17:15.8033271Z   CMake Warning (dev) at /home/conda/feedstock_root/build_artifacts/pytorch-recipe_1627840968659/_build_env/share/cmake-3.21/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
2021-08-01T18:17:15.8038217Z     The package name passed to `find_package_handle_standard_args` (NCCL) does
2021-08-01T18:17:15.8042974Z     not match the name of the calling package (nccl).  This can lead to
2021-08-01T18:17:15.8047842Z     problems in calling code that expects `find_package` result variables
2021-08-01T18:17:15.8053787Z     (e.g., `_FOUND`) to follow a certain pattern.
2021-08-01T18:17:15.8058581Z   Call Stack (most recent call first):
2021-08-01T18:17:15.8063202Z     third_party/gloo/cmake/Modules/Findnccl.cmake:45 (find_package_handle_standard_args)
2021-08-01T18:17:15.8069013Z     third_party/gloo/cmake/Dependencies.cmake:128 (find_package)
2021-08-01T18:17:15.8075112Z     third_party/gloo/CMakeLists.txt:109 (include)
2021-08-01T18:17:15.8080689Z   This warning is for project developers.  Use -Wno-dev to suppress it.
2021-08-01T18:17:15.8086418Z 
2021-08-01T18:17:15.8095830Z   -- Determining NCCL version from the header file: /home/conda/feedstock_root/build_artifacts/pytorch-recipe_1627840968659/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include/nccl.h
2021-08-01T18:17:15.8103912Z   -- NCCL_MAJOR_VERSION: 2
2021-08-01T18:17:15.8112940Z   -- Found NCCL (include: /home/conda/feedstock_root/build_artifacts/pytorch-recipe_1627840968659/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include, library: /home/conda/feedstock_root/build_artifacts/pytorch-recipe_1627840968659/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/lib/libnccl.so)
```

Reply:
Thanks @IvanYashchuk for the context and chiming in!